features/upgrade/sdn/policy-upgrade.feature

```diff
@@ -477,3 +477,173 @@ Feature: SDN compoment upgrade testing
+      | {"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"policyAuditConfig": {"destination": "udp:<%= cb.svc_ip %>:514" }}}}} |
+    And I wait up to 30 seconds for the steps to pass:
+    """
+      Given the status of condition "Progressing" for network operator is :True
+    """
+    And I wait up to 300 seconds for the steps to pass:
+    """
+      Given the status of condition "Progressing" for network operator is :False
+      And the status of condition "Available" for network operator is :True
+    """
+
+    When I run the :new_project client command with:
+      | project_name | policy-upgrade6 |
+    Then the step should succeed
+    And evaluation of `"policy-upgrade6"` is stored in the :proj1 clipboard
+    When I run the :annotate admin command with:
+      | resource     | namespace                                                |
+      | resourcename | <%= cb.proj1 %>                                          |
+      | keyval       | k8s.ovn.org/acl-logging={"deny":"alert","allow":"alert"} |
+    Then the step should succeed
+    When I run the :new_project client command with:
+      | project_name | policy-upgrade7 |
+    Then the step should succeed
+    And evaluation of `"policy-upgrade7"` is stored in the :proj2 clipboard
+    When I run the :label admin command with:
+      | resource | namespace       |
+      | name     | <%= cb.proj2 %> |
+      | key_val  | team=operations |
+    Then the step should succeed
+    Given I use the "<%= cb.proj1 %>" project
+    Given I obtain test data file "networking/list_for_pods.json"
+    When I run oc create over "list_for_pods.json" replacing paths:
+      | ["items"][0]["spec"]["replicas"] | 1 |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(1).ip_url` is stored in the :p1pod1ip clipboard
+    Given I obtain test data file "networking/networkpolicy/allow-ns-and-pod.yaml"
+    When I run the :create admin command with:
+      | f | allow-ns-and-pod.yaml |
+      | n | <%= cb.proj1 %>       |
+    Then the step should succeed
+    Given I use the "<%= cb.proj2 %>" project
+    Given I obtain test data file "networking/list_for_pods.json"
+    When I run oc create over "list_for_pods.json" replacing paths:
+      | ["items"][0]["spec"]["replicas"] | 1 |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(2).name` is stored in the :p2pod1name clipboard
+    Given I obtain test data file "rc/idle-rc-1.yaml"
+    When I run oc create over "idle-rc-1.yaml" replacing paths:
+      | ["items"][0]["spec"]["replicas"] | 1 |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=hello-idle |
+    And evaluation of `pod(3).name` is stored in the :p2pod2name clipboard
+    And I wait up to 20 seconds for the steps to pass:
```

Comment:
@asood-rh If you think particular step in this block expects a wait, please add there. With 20 seconds wait for whole block and couple of --connect-timeouts, doesn't making sense at this point IMO.

Reply:
@anuragthehatter Need the wait as there are network policies that are added to the project. Need the --connect-timeout so that curl requests terminate after specified time where curl is expected to fail because of policy.
I have removed the -connect-timeout where curl request will succeed.

Reply:
@asood-rh The flows typically takes milliseconds to be effective. I still suggest keeping timeouts in both cases for at least 2 seconds. You are creating a pod at line #559 (which will trigger some ovn flows) and immediately a curl after that without timeout. Your curl might get lucky sometimes or not depending on platforms configuration/cpu/mem etc.

As per my understanding you need a specific time loop at 572-574 and 582-584 only. And connect timeouts of 2s should suffice in both cases as discussed.

Reply:
@anuragthehatter  So now both --connect-timeout are fine but need two time loops? Would not add double the time in script? Please clarify.

Reply:
>So now both --connect-timeout are fine

Suggesting --connect-timeout 2 at least in both #568 and #578 based on above discussion

> Would not add double the time in script

That seems logically right Its a sequential execution flow as we know. If we are putting connect-timeouts wait under another timer loop,  We are kinda creating time { time } loop. If you think /var/log will take 20 seconds to populate keyword you are looking for, its okay to loop that for 20s (Thats kinda would be strange time in load tests) Same for following /val/log. Just my 3 cents :)

Reply:
Sure.Thanks @asood-rh  And I also meant following logic for both /var/log steps. Its good idea to keep them separated as discussed above. It would also make reviewer understand that 20 second is for /var/log steps but not for curl loops.

``` 
And I wait up to 20 seconds for the steps to pass:
    """
        When I execute on the "<%= cb.rsyslog_server_pod %>" pod:
      | grep | -w | verdict=allow | /var/log/messages |
    Then the output should contain "verdict=allow"
    """
```

Reply:
@anuragthehatter As discussed earlier breaking the block into two blocks adds more time to the script and I do want retry the curl if it fails the first time in the given time for the steps to pass.

Reply:
We are creating normal pods across 2 projects incorporating an allow-ns-and-pod policy. If the curl expects to work in upto 20 seconds across the pods, its definitely a bug as this will get worse in perf test iterations. Also breaking the blocks won't increase the script time but should be same 20 seconds or less in PASS'ing situation (if you have waited 20 sec for verdict=drop, verdict=allow would also populate by that time so next 20 sec wait wont wait fully). In fail case, which we expect once in a while it might go up to 40 seconds and that will help rule out issue easily bte curls vs syslog.

Anyways leaving upto you to decide, just a suggestion semantically..

Reply:
@anuragthehatter Two blocks as two set of steps are not related but retries needed for curl as well as the syslog log messages.